docs/docs/aleph_zero/types.md

```diff
@@ -4,58 +4,79 @@ title: Types
 slug: /aleph_zero/types
 ---
 
+This segment introduces key data types integral to understanding and working with the Aleph Zero protocol. These types play a pivotal role in expressing and managing various aspects of the protocol's functionality.
 
-|Name|Description|
-|-|-|
-|SqrtPrice|Square root of price value|
-|FeeGrowth|Accumulated amount of fees|
-|FixedPoint|Number with fixed number of decimal places|
-|Liquidity|Amount of virtual liquidity|
-|Percentage|Represents values as fractions of 100|
-|SecondsPerLiquidity|Measures the time inside/outside liquidity|
-|TokenAmount|Quantity of specific token|
+### Defining Decimal:
+We have implemented a custom decimal system, which is detailed in our repository [here](https://github.com/invariant-labs/decimal). The structure of the decimal is outlined below in Rust syntax:
+```rust
+#[decimal(#scale, #big_type)]
+pub struct DecimalName {
+  pub v: #underlying_type
+}
+```
+- **#scale**: An `integer` that determines the number of decimal places.
+- **#big_type**: The type to which it will be extended in intermediate operations (default is U256).
+- **DecimalName**: The name of the struct.
+- **#underlying_type**: The underlying numeric type.
```

Comment:
You can also add some example:
e.g. the value 12042 with decimal 3 should be interpreted as: 12042 * 10^(-3)